au4/src/record/internal/au3/au3record.cpp

```diff
@@ -241,21 +241,46 @@ void Au3Record::init()
     });
 
     s_recordingListener->recordingClipChanged().onReceive(this, [this](const trackedit::ClipKey& clipKey) {
-        Track* track = &PendingTracks::Get(projectRef())
-                       .SubstitutePendingChangedTrack(*DomAccessor::findWaveTrack(projectRef(), TrackId(clipKey.trackId)));
+        WaveTrack* origWaveTrack = DomAccessor::findWaveTrack(projectRef(), TrackId(clipKey.trackId));
 
-        WaveTrack* waveTrack = dynamic_cast<WaveTrack*>(track);
-        IF_ASSERT_FAILED(waveTrack) {
+        Track* pendingTrack = &PendingTracks::Get(projectRef())
+                       .SubstitutePendingChangedTrack(*origWaveTrack);
+
+        WaveTrack* pendingWaveTrack = dynamic_cast<WaveTrack*>(pendingTrack);
+        IF_ASSERT_FAILED(pendingWaveTrack) {
+            return;
+        }
+
+        // Iterate through both tracks simultaneously to find matching clip
+        // in the original track and get its counterpart in the pending track.
+        std::optional<uint64_t> clipIdToUpdate;
+        auto origIter = origWaveTrack->Intervals().begin();
+        auto pendingIter = pendingWaveTrack->Intervals().begin();
+
+        for (; origIter != origWaveTrack->Intervals().end() && pendingIter != pendingWaveTrack->Intervals().end(); ++origIter, ++pendingIter) {
+            if (au::au3::WaveClipID((*(origIter)).get()).id == clipKey.clipId) {
+                clipIdToUpdate = au::au3::WaveClipID((*(pendingIter)).get()).id;
+                break;
+            }
+        }
+
+        if (!clipIdToUpdate) {
             return;
         }
 
-        std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.clipId);
+        std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(pendingWaveTrack, clipIdToUpdate.value());
         IF_ASSERT_FAILED(clip) {
             return;
         }
 
         trackedit::ITrackeditProjectPtr prj = globalContext()->currentTrackeditProject();
-        prj->onClipChanged(DomConverter::clip(waveTrack, clip.get()));
+
+        // IMPORTANT: getting pending clip and assigning it to the clip from original track
+        // to make that possible we need to assign orignal clip's ID to the pending clip
+        // so onClipChanged accepts it
+        auto pendingClipWithFakeId = DomConverter::clip(pendingWaveTrack, clip.get());
+        pendingClipWithFakeId.key.clipId = clipKey.clipId;
+        prj->onClipChanged(pendingClipWithFakeId);
```

Comment:
The idea is clear... but I don't like what's happening here (
I need to think about it...
Let's keep it like this for now